src/widget.c

```diff
@@ -216,20 +286,28 @@ extern void led_process_thread(void *d0, void *d1, void *d2) {
                 blink.duration_ms);
 
         // turn appropriate LEDs on
-        for (uint8_t pos = 0; pos < 3; pos++) {
-            if (BIT(pos) & blink.color) {
-                led_on(led_dev, rgb_idx[pos]);
-            }
+        if (blink.color == current_color && current_color != 0) {
+            set_rgb_leds(current_color, 0);
+            k_sleep(K_MSEC(blink.duration_ms));
+            set_rgb_leds(0, blink.color);
+        } else {
+            set_rgb_leds(current_color, blink.color);
         }
 
         // wait for blink duration
         k_sleep(K_MSEC(blink.duration_ms));
 
+#if SHOW_LAYER_COLORS
+        current_color = led_current_layer_color;
+#endif
+
         // turn appropriate LEDs off
-        for (uint8_t pos = 0; pos < 3; pos++) {
-            if (BIT(pos) & blink.color) {
-                led_off(led_dev, rgb_idx[pos]);
-            }
+        if (blink.color == current_color && current_color != 0) {
+            set_rgb_leds(blink.color, 0);
+            k_sleep(K_MSEC(blink.duration_ms));
```

Comment:
Maybe we can use this value instead?
```suggestion
            k_sleep(K_MSEC(CONFIG_RGBLED_WIDGET_INTERVAL_MS));
```